x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -103,24 +110,23 @@ public EsqlResponseListener(RestChannel channel, RestRequest restRequest, EsqlQu
         this.channel = channel;
         this.restRequest = restRequest;
         this.esqlQuery = esqlRequest.query();
-        mediaType = EsqlMediaTypeParser.getResponseMediaType(restRequest, esqlRequest);
+        this.mediaType = EsqlMediaTypeParser.getResponseMediaType(restRequest, esqlRequest);
 
-        /*
-         * Special handling for the "delimiter" parameter which should only be
-         * checked for being present or not in the case of CSV format. We cannot
-         * override {@link BaseRestHandler#responseParams()} because this
-         * parameter should only be checked for CSV, not other formats.
-         */
-        if (mediaType != CSV && restRequest.hasParam(URL_PARAM_DELIMITER)) {
-            String message = String.format(
-                Locale.ROOT,
-                "parameter: [%s] can only be used with the format [%s] for request [%s]",
-                URL_PARAM_DELIMITER,
-                CSV.queryParameter(),
-                restRequest.path()
-            );
-            throw new IllegalArgumentException(message);
-        }
+        checkDelimiter();
+    }
+
+    /**
+     * Async query get API does not have an esqlRequest and mediaType defaults to JSON if no "format" in params
+     */
+    public EsqlResponseListener(RestChannel channel, RestRequest restRequest) {
+        super(channel);
+
+        this.channel = channel;
+        this.restRequest = restRequest;
+        this.esqlQuery = null;
```

Comment:
The `esqlQuery` is used for logging -- an altnative to making this `null` is to have an `esqlQueryOrId` that could also store the async ID it's being asked for.